termsandconditions/middleware.py

```diff
@@ -31,8 +33,21 @@ def process_request(self, request):
         current_path = request.META['PATH_INFO']
 
         if request.user.is_authenticated() and is_path_protected(current_path):
-            for term in TermsAndConditions.get_active_list():
-                if not TermsAndConditions.agreed_to_latest(request.user, term):
+            active_terms = (
+                TermsAndConditions.objects.prefetch_related('users')
+                .filter(
+                    date_active__isnull=False,
+                    date_active__lte=timezone.now(),
+                )
+                .values_list('slug')
+                .annotate(Max('date_active'))
+                .order_by()
```

Comment:
Hi - I'm not clear what the annotate(Max) and the order_by() with no arguments are doing here...?  Is the goal to present the terms in a particular order for the user to agree to them?

Reply:
The `order_by` came forth from [this question](https://stackoverflow.com/questions/46898464/django-query-group-by-field-and-get-the-latest-entry-per-group) and it has to do with this [Django detail](https://docs.djangoproject.com/en/dev/topics/db/aggregation/#interaction-with-default-ordering-or-order-by). Otherwise, the results of the query are not correct.